docs/hugo/content/design/ADR-2023-05-Installing-Only-Selected-CRDs.md

```diff
@@ -0,0 +1,248 @@
+
+Pros:
+* More structured than a `ConfigMap`.
+
+Cons:
+* Chicken and egg problem of using the CRD that's part of the operator to configure the operator behavior.
+  Helm has a very hard time creating a CRD _and_ an instance of the CRD in the same deployment.
+* May need to implement collection strategic merge behavior that already exists for `ConfigMap`.
+
+### Question 3: How do we ensure consistent configuration as ASO is upgraded
+
+#### Answer 3a: Require the user to be consistent when upgrading
+
+The user would need re-specify what resources they want every time they upgrade
+
+Pros:
+* Easy for us
+
+Cons:
+* Very easy to make a mistake and accidentally leave a resource off they want.
+
+#### Answer 3b: Use a configuration technique that persists across upgrades
+
+Pros:
+* Satisfies the requirement.
+
+Cons:
+* Would augment/complicate the approach we already have
+* More complex for us to implement, especially with Helm. Helm owns any resource it creates, 
+  and deletes it if not specified on the next upgrade. See [Using a ConfigMap](#answer-2b--use-a-configmap) for some 
+  workarounds.
+
+#### Answer 3c: Query existing CRDs and always support those
+
+Combined with some form of non-persistent storage such as [command line](#answer-2a--using-the-existing-command-line-parameter)
+we would query all existing CRDs and install/upgrade them, while adding whatever new CRDs were specified in 
+the non-persistent storage.
+
+Pros:
+* Automatically supports seamless upgrades.
+* Installed resources are always upgrade to the latest version, even if user has forgotten about them.
+* Could be built onto existing cmdline configuration.
+* Hard for users to get wrong.
+
+Cons:
+* ???
+
+## Decision
+
+### Answer to [question #1: What to install by default?](#question-1--what-to-install-by-default)
+
+A combination of  [Answer 1b: Nothing by default](#answer-1b--nothing-by-default) and 
+[Answer 1c: Mandatory configuration](#answer-1c--mandatory-configuration).
+
+Specifically, the operator pod should exit with an error if there are no CRDs configured in `--crd-pattern` _and_ no
+existing CRDs in the cluster. If a `--crd-pattern` is specified, or if there are already existing ASO CRDs in the cluster
+then the pod will launch without an error, install/upgrade the required CRDs and then start watches on those CRDs.
+
+### Answer to [question #2: How do users specify what CRDs to install?](#question-2--how-do-users-specify-what-crds-to-install)
+
+[Answer 2a: Using the existing command line parameter](#answer-2a--using-the-existing-command-line-parameter)
+
+This parameter will come to mean: "New CRDs to install to the cluster".
+
+### Answer to [question #3: How do we ensure consistent configuration as ASO is upgraded?](#question-3--how-do-we-ensure-consistent-configuration-as-aso-is-upgraded)
+
+[Answer 3c: Query existing CRDs and always support those](#answer-3c--query-existing-crds-and-always-support-those)
```

Comment:
Note: We need to deal with the users-manage-their-own CRDs case, explain what we do for that here.

Reply:
Confirm that in the "we dont manage CRDs case, we still have GET CRDs permissions"

Reply:
Updated documentation to cover this case. Fundamentally, it should work exactly as planned. See the document which outlines the basic algorithm